src/components/IOUConfirmationList.js

```diff
@@ -165,63 +216,54 @@ class IOUConfirmationList extends Component {
         if (!this.props.hasMultipleParticipants) {
             return null;
         }
-
-        const splits = this.props.participants.map(participant => ({
+        const selectedParticipants = this.getSelectedParticipants();
+        const splits = selectedParticipants.map(participant => ({
             email: participant.login,
 
             // We should send in cents to API
             // Cents is temporary and there must be support for other currencies in the future
-            amount: this.calculateAmount(),
+            amount: this.calculateAmount(selectedParticipants),
         }));
 
         splits.push({
             email: this.props.myPersonalDetails.login,
 
             // The user is default and we should send in cents to API
             // USD is temporary and there must be support for other currencies in the future
-            amount: this.calculateAmount(true),
+            amount: this.calculateAmount(selectedParticipants, true),
         });
         return splits;
     }
 
     /**
-     * Gets participants list for a report
-     *
+     * Returns selected options -- there is checkmark for every row in List for split flow
      * @returns {Array}
      */
-    getParticipants() {
-        const participants = this.props.participants.map(participant => participant.login);
-        participants.push(this.props.myPersonalDetails.login);
-        return participants;
-    }
-
-    /**
-     * Returns selected options with all participant logins -- there is checkmark for every row in List for split flow
-     * @returns {Array}
-     */
-    getAllOptionsAsSelected() {
+    getSelectedOptions() {
         if (!this.props.hasMultipleParticipants) {
             return [];
         }
+        const selectedParticipants = this.getSelectedParticipants();
         return [
-            ...this.props.participants,
+            ...selectedParticipants,
             getIOUConfirmationOptionsFromMyPersonalDetail(this.props.myPersonalDetails),
         ];
     }
 
     /**
-     * Calculates the amount per user
+     * Calculates the amount per user given a list of participants
+     * @param {Array} participants
      * @param {Boolean} isDefaultUser
      * @returns {Number}
      */
-    calculateAmount(isDefaultUser = false) {
+    calculateAmount(participants, isDefaultUser = false) {
```

Comment:
It looks like every time we call this function we pass the selected participants for this parameter. I think we can just access the selected participants inside of this function rather than passing them in every time. What do you think?

Reply:
Yes, we always pass selected participants.
Will remove the parameter.

Reply:
Found a single exception. 
`getParticipantsWithAmount()` passes `props.participants` to `calculateAmount()` when called from the constructor.

Reply:
Oh, right, so this.state.participants wouldn't exist yet, good point 👍